### PR TITLE
yamllint all the things

### PR DESCRIPTION
**Pull request by Marwel**
Hi,
created a yamllint config, that should be reasonable and a github action, so that future contribs are at least valid yaml.
Also fixed all the errors and warnings, that yamllint wanted a fix for.

### CODE REVIEW

File: docs/configuring-playbook-bot-go-neb.md [hunk 1]
```diff
@@ -198,8 +198,8 @@ matrix_bot_go_neb_services:
       # Each room will get the notification with the alert rendered with the given template
       rooms:
         "!someroomid:domain.tld":
-          text_template: "{% raw %}{{range .Alerts -}} [{{ .Status }}] {{index .Labels \"alertname\" }}: {{index .Annotations \"description\"}} {{ end -}}{% endraw %}"
-          html_template: "{% raw %}{{range .Alerts -}}  {{ $severity := index .Labels \"severity\" }}    {{ if eq .Status \"firing\" }}      {{ if eq $severity \"critical\"}}        <font color='red'><b>[FIRING - CRITICAL]</b></font>      {{ else if eq $severity \"warning\"}}        <font color='orange'><b>[FIRING - WARNING]</b></font>      {{ else }}        <b>[FIRING - {{ $severity }}]</b>      {{ end }}    {{ else }}      <font color='green'><b>[RESOLVED]</b></font>    {{ end }}  {{ index .Labels \"alertname\"}} : {{ index .Annotations \"description\"}}   <a href=\"{{ .GeneratorURL }}\">source</a><br/>{{end -}}{% endraw %}"
+          text_template: "{% raw %}{{range .Alerts -}} [{{ .Status }}] {{index .Labels \"alertname\"}}: {{index .Annotations \"description\"}} {{ end -}}{% endraw %}"
```
Comment (spantaleev) [inline docs/configuring-playbook-bot-go-neb.md:201]: Looks like `yamllint`:

- insists on removing trailing space from "Jinja2" statements (`{{ if eq .Status \"firing\" }}` -> `{{ if eq .Status \"firing\"}}`)
- operates within `{% raw %}` blocks. This thing here is a Go template, not a Jinja2 print statement

File: docs/configuring-playbook-bot-mjolnir.md [hunk 2]
```diff
@@ -28,7 +28,7 @@ If you use curl, you can get an access token like this:
 
 ```
 curl -X POST --header 'Content-Type: application/json' -d '{
-    "identifier": { "type": "m.id.user", "user": "bot.mjolnir" },
+    "identifier": { "type": "m.id.user", "user": "bot.mjolnir"},
```
Comment (spantaleev) [inline docs/configuring-playbook-bot-mjolnir.md:31]: Same here - leading space in the `{` block, without trailing space before `}`.

Perhaps removing the leading space will also make yamllint happy?

Comment (Marwel) [reply docs/configuring-playbook-bot-mjolnir.md:31]: That could be my mistake. I did not want to edit all these changes by myself, so in the end i did most of these changes via find and replace.

**Comment by Marwel** [comment]
Ok, i reverted the changes to the md files.

**Comment by skepticalwaves** [comment]
Why not https://github.com/marketplace/actions/docker-lint ?

**Comment by Marwel** [comment]
This PR is only about yamllint, most of this repo contains yaml files.
Your mentioned action lints Dockerfiles, so that is another issue.

**Comment by spantaleev** [comment]
Looking at this PR, it looks like it still makes various things uglier. Example:

```diff
-matrix_beeper_linkedin_bridge_presence: "{{ matrix_synapse_presence_enabled if matrix_synapse_enabled else true }}"
+matrix_beeper_linkedin_bridge_presence: "{{ matrix_synapse_presence_enabled if matrix_synapse_enabled else true}}"
```

**Comment by Marwel** [comment]
I'm gonna have a look at those spaces again.

**Comment by Marwel** [comment]
Ok, i fixed those missing spaces.
I know this PR looks massive, but it should mostly be indentation, truthies and some spaces.

File: roles/matrix-synapse/vars/workers.yml [hunk 3]
```diff
@@ -206,7 +206,7 @@ matrix_synapse_workers_generic_worker_endpoints:
   # You might also wish to investigate the `update_user_directory` and
   # `media_instance_running_background_jobs` settings.
 
-# pusher worker (no API endpoints) [
+  # pusher worker (no API endpoints) [
```
Comment (spantaleev) [inline roles/matrix-synapse/vars/workers.yml]: This file is automatically generated (see `roles/matrix-synapse/files/workers-doc-to-yaml.sh`).

The next time we run that file to regenerate `workers.yml`, these changes will be thrown out.

**Comment by spantaleev** [comment]
It is massive, indeed! Thank you for spending all that time on this! :+1: 

I've fixed up a few things and merged this as it is. We will probably need to fix the `roles/matrix-synapse/vars/workers.yml` situation though.

**Comment by Marwel** [comment]
I just "fixed" this more or less with #1613.

**Pull request merged**